### PR TITLE
Pass plan environment

### PR DESCRIPTION
**Pull request by fernflower**
Prior to this change environment variables declared via in plan's
environment were ignored.

### CODE REVIEW

File: tmt/steps/provision/__init__.py [hunk 1]
```diff
@@ -362,7 +362,9 @@ def _export_environment(self, execute_environment=None):
     def ansible(self, playbook):
         """ Prepare guest using ansible playbook """
         playbook = self._ansible_playbook_path(playbook)
+        environment = self._export_environment()
         stdout, stderr = self.run(
+            f'{environment}'
```
Comment (psss) [inline tmt/steps/provision/__init__.py:367]:
```suggestion
            f'{self._export_environment()}'
```
Let's use the method directly.

**Review by psss — CHANGES_REQUESTED** [review]
Would be nice to include a corresponding test coverage. What about extending a bit the `/tests/prepare/ansible` test? The commit message has a typo. Let's not use prefixes in the commit summary as suggested in the [commit recommendations](https://tmt.readthedocs.io/en/latest/contribute.html#commits). Thanks.

**Comment by psss** [comment]
> That's critical for leapp e2e tests to run properly with tmt

Marking as a `must have` for `1.6`.

**Comment by fernflower** [comment]
> > That's critical for leapp e2e tests to run properly with tmt

@psss remember when I said, that I was wrong and it isn't critical as I found a workaround? Well turns out I didn't and it is blocking e2e test transition for real. 

Added tests, F34 job is failing though :(

**Review by psss — APPROVED** [review]
Thanks for providing the test coverage. During the review I realized that the `local` provision has the same problem so I've covered that in bb68cad together with extending the test coverage to actually verify `local` and `virtual` provision methods as well.

**Review by lukaszachy — APPROVED** [review]

**Pull request merged**